src/windows/FileTransferProxy.js

```diff
@@ -108,13 +108,15 @@ function doUpload (upload, uploadId, filePath, server, successCallback, errorCal
                     // in the other way, try to get response property
                     var response = upload.getResponseInformation();
                     if (!response) {
-                        resolve(new FTErr(FTErr.CONNECTION_ERR, source, server));
+                        resolve(new FTErr(FTErr.CONNECTION_ERR, source, server, null, null, error));
                     } else {
                         var reader = new Windows.Storage.Streams.DataReader(upload.getResultStreamAt(0));
-                        reader.loadAsync(upload.progress.bytesReceived).then(function (size) {
+                        reader.loadAsync(upload.progress.bytesReceived).done(function (size) {
```

Comment:
Where does the `done` method come from. I can't seem to find any documentation on it?

Reply:
The docs are here: https://docs.microsoft.com/en-us/previous-versions/windows/apps/hh701079(v=win.10) . But I actually saw you can also use the second parameter for then which I didn't know. Let me use that instead.

Reply:
https://docs.microsoft.com/en-us/previous-versions/windows/apps/hh700339(v=win.10)

Reply:
I actually see that they use `done` in line 478 including the error handling. I have tried that `then` before when I was doing the server settings file and no success but it worked with `done`.

Reply:
ok changed it back to then. I checked the code and it has a second argument for onError